### PR TITLE
Follow up from #2452

### PR DESCRIPTION
**Pull request by Guardiola31337**
Follow up from https://github.com/mapbox/mapbox-navigation-android/pull/2452

Addressing 👇 

- > We can merge whenever a new Nav Native snapshot is released (and bumped here) with both `key_points` and `uncertain` state changes.
- https://github.com/mapbox/mapbox-navigation-android/pull/2452#issuecomment-583567113 (This is based now on https://github.com/mapbox/mapbox-gl-native-android/commit/24ed3be62696ee67fcdfc0a2734283d0f1f72efb from Maps SDK)
- https://github.com/mapbox/mapbox-navigation-android/pull/2452#issuecomment-583656993

One thing that I noticed though is that we're still cutting corners 🤔 

lat_long_workaround_removed

Will keep 👀 

cc @mskurydin @zugaldia @andrewychen @Aurora-Boreal @LukasPaczos @SiarheiFedartsou

### CODE REVIEW

**Comment by mskurydin** [comment]
Yes, in this case we were in uncertain state and cutting corners works only for snapping to a route. Supporting this for free drive requires massive changes and will be automatically resolved when we implement eHorizon for BMW (I think).

**Comment by Guardiola31337** [comment]
@mskurydin 
> Yes, in this case we were in uncertain state and cutting corners works only for snapping to a route.

We're in _Active Guidance_ . though 🤔

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@ ext {
       mapboxSdkDirectionsModels : '5.1.0-beta.1',
       mapboxEvents              : '4.5.1',
       mapboxCore                : '1.3.0',
-      mapboxNavigator           : 'sf-keypoints-SNAPSHOT-8',
+      mapboxNavigator           : 'master-SNAPSHOT-0',
```
Comment (LukasPaczos) [inline gradle/dependencies.gradle]: is this the most up-to-date snapshot @mskurydin @Guardiola31337?

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt [hunk 2]
```diff
@@ -101,7 +101,11 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
 
     // Offline
 
-    override fun configureRouter(routerParams: RouterParams, httpClient: HttpInterface): Long =
+    override fun cacheLastRoute() {
```
Comment (LukasPaczos) [inline libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt]: Should we expose a `NavigationOptions` boolean flag that determines whether we call this method or not after we set the route? We can set it up in a following PR as well.

File: gradle/dependencies.gradle (hunk 1 above)
Comment (Guardiola31337) [reply gradle/dependencies.gradle]: Actually not, `ms-bearing-from-shape-on-interpolation-SNAPSHOT-2` is the most up to date. Will bump.

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt (hunk 2 above)
Comment (Guardiola31337) [reply libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt]: So you mean not exposing this method at all and add a flag to `setRoute` (via `NavigationOptions`) and control if `cacheLastRoute` should be called or not? If so, yeah that's an option, we can also control that upstream in `MapboxTripSession` / `MapboxNavigation` (`MapboxTripSession` doesn't accept `NavigationOptions` but I believe it's going to be needed at some point). Is that what you were thinking? In any case, we can follow up in a separate PR as you mentioned and not block this one 👍

Comment (LukasPaczos) [reply libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt]: One of those works, we can either pass the options or call from upstream. Let's decide in the follow up :+1:

**Pull request merged**